lib/ansible/modules/windows/win_domain_user.ps1

```diff
@@ -0,0 +1,271 @@
+
+# Module control parameters
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","query"
+$update_password = Get-AnsibleParam -obj $params -name "update_password" -type "str" -default "always" -validateset "always","on_create"
+$groups_action = Get-AnsibleParam -obj $params -name "groups_action" -type "str" -default "replace" -validateset "add","remove","replace"
+
+# User account parameters
+$username = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$description = Get-AnsibleParam -obj $params -name "description" -type "str"
+$password = Get-AnsibleParam -obj $params -name "password" -type "str"
+$password_expired = Get-AnsibleParam -obj $params -name "password_expired" -type "bool"
+$password_never_expires = Get-AnsibleParam -obj $params -name "password_never_expires" -type "bool"
+$user_cannot_change_password = Get-AnsibleParam -obj $params -name "user_cannot_change_password" -type "bool"
+$account_locked = Get-AnsibleParam -obj $params -name "account_locked" -type "bool"
+$groups = Get-AnsibleParam -obj $params -name "groups" -type "list"
+$enabled = Get-AnsibleParam -obj $params -name "enabled" -type "bool" -default $true
+$path = Get-AnsibleParam -obj $params -name "path" -type "str"
+$upn = Get-AnsibleParam -obj $params -name "upn" -type "str"
+
+# User informational parameters
+$user_info = @{
+    GivenName = Get-AnsibleParam -obj $params -name "firstname" -type "str"
+    Surname = Get-AnsibleParam -obj $params -name "surname" -type "str"
+    Company = Get-AnsibleParam -obj $params -name "company" -type "str"
+    EmailAddress = Get-AnsibleParam -obj $params -name "email" -type "str"
+    StreetAddress = Get-AnsibleParam -obj $params -name "street" -type "str"
+    City = Get-AnsibleParam -obj $params -name "city" -type "str"
+    State = Get-AnsibleParam -obj $params -name "state_province" -type "str"
+    PostalCode = Get-AnsibleParam -obj $params -name "postal_code" -type "str"
+    Country = Get-AnsibleParam -obj $params -name "country" -type "str"
+}
+
+# Parameter validation
+If ($account_locked -ne $null -and $account_locked) {
+    Fail-Json $result "account_locked must be set to 'no' if provided"
+}
+If (($password_expired -ne $null) -and ($password_never_expires -ne $null)) {
+    Fail-Json $result "password_expired and password_never_expires are mutually exclusive but have both been set"
+}
+
+try {
+    $user_obj = Get-ADUser -Identity $username -Properties *
+}
+catch {
+    $user_obj = $null
+}
+
+If ($state -eq 'present') {
+    # Ensure user exists
+    try {
+        $new_user = $false
+
+        # If the account does not exist, create it
+        If (-not $user_obj) {
+            If ($path -ne $null){
+                New-ADUser -Name $username -Path $path
+            }
+            Else {
+                New-ADUser -Name $username
+            }
+            $user_obj = Get-ADUser -Identity $username -Properties *
```

Comment:
I guess, skip everything that follows on $check_mode (e.g. by running Exit-Json)